lib/widgets/message_list.dart

```diff
@@ -523,11 +532,58 @@ class MarkAsReadWidget extends StatelessWidget {
                     height: (23 / 18))
                   .merge(weightVariableTextStyle(context, wght: 400))),
                 shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(7)),
+              ).copyWith(
+                // clobber `FilledButton`'s default disabled state
+                // TODO(#95) need dark-theme colors (foreground and background)
+                foregroundColor: WidgetStateColor.resolveWith((_) => Colors.white),
+                backgroundColor: WidgetStateColor.resolveWith((_) => _UnreadMarker.color),
               ),
-              onPressed: () => _handlePress(context),
+              onPressed: loading ? null : () => _handlePress(context),
               icon: const Icon(Icons.playlist_add_check),
-              label: Text(zulipLocalizations.markAllAsReadLabel))))),
-    );
+              label: Text(zulipLocalizations.markAllAsReadLabel))))));
+  }
+}
+
+class MarkAsReadAnimation extends StatefulWidget {
+  final bool loading;
+  final bool hidden;
+  final Widget child;
+
+  const MarkAsReadAnimation({
+    super.key,
+    required this.loading,
+    required this.hidden,
+    required this.child
+  });
+
+  @override
+  State<MarkAsReadAnimation> createState() => _MarkAsReadAnimationState();
+}
+
+class _MarkAsReadAnimationState extends State<MarkAsReadAnimation> {
+  bool _isPressed = false;
+
+  void _setScale(bool isPressed) {
+    setState(() {
+      _isPressed = isPressed;
```

Comment:
nit: this method's name doesn't reflect what it does; it also makes it hard to guess what the meaning of an argument "true" vs. "false" is. A good name would be `_setIsPressed`.